butterfree/load/writers/delta_feature_store_writer.py

```diff
@@ -0,0 +1,179 @@
+        merge_on: list of columns to use as merge keys.
+        deduplicate: whether to deduplicate data before merging based on featr set keys.
+            Default is False.
+        when_not_matched_insert: optional condition for insert operations.
+            When provided, rows will only be inserted if this condition is true.
+        when_matched_update: optional condition for update operations.
+            When provided, rows will only be updated if this condition is true.
+            Source columns can be referenced as source.<column_name> and target
+            columns as target.<column_name>.
+        when_matched_delete: optional condition for delete operations.
+            When provided, rows will be deleted if this condition is true.
+            Source and target columns can be referenced as in update conditions.
+
+    Example:
+        Simple example regarding DeltaFeatureStoreWriter class instantiation.
+        We can instantiate this class with basic merge configuration:
+
+    >>> from butterfree.load.writers import DeltaFeatureStoreWriter
+    >>> spark_client = SparkClient()
+    >>> writer = DeltaFeatureStoreWriter(
+    ...     database="feature_store",
+    ...     table="user_features",
+    ...     merge_on=["id", "timestamp"]
+    ... )
+    >>> writer.write(feature_set=feature_set,
+    ...             dataframe=dataframe,
+    ...             spark_client=spark_client)
+
+        We can also enable deduplication based on the feature set keys:
+
+    >>> writer = DeltaFeatureStoreWriter(
+    ...     database="feature_store",
+    ...     table="user_features",
+    ...     merge_on=["id", "timestamp"],
+    ...     deduplicate=True
+    ... )
+
+        For more control over the merge operation, we can add conditions:
+
+    >>> writer = DeltaFeatureStoreWriter(
+    ...     database="feature_store",
+    ...     table="user_features",
+    ...     merge_on=["id", "timestamp"],
+    ...     when_matched_update="source.value > target.value",
+    ...     when_not_matched_insert="source.value > 0"
+    ... )
+
+        The writer supports schema evolution by default and will automatically
+        handle updates to the feature set schema.
+
+        When writing with deduplication enabled, the writer will use the feature
+        set's key columns and timestamp to ensure data quality by removing
+        duplicates before merging.
+
+        For optimal performance, it's recommended to:
+        1. Choose appropriate merge keys
+        2. Use conditions to filter unnecessary updates/inserts
+        3. Enable deduplication only when needed
+    """
+
+    def __init__(
+        self,
+        database: str,
+        table: str,
+        merge_on: List[str],
+        deduplicate: bool = False,
+        when_not_matched_insert: Optional[str] = None,
+        when_matched_update: Optional[str] = None,
+        when_matched_delete: Optional[str] = None,
+    ):
+        self.config = DeltaConfig(
```

Comment:
Was wondering if we should reuse keep the standard of having writer configs as concrete classes of the [AbstractWriterConfig](https://github.com/quintoandar/butterfree/blob/staging/butterfree/configs/db/abstract_config.py). I understand that we wouldn't reuse much of the enforced methods there. Any thoughts on that?